signer/src/block_observer.rs

```diff
@@ -267,19 +268,26 @@ where
     #[tracing::instrument(skip_all, fields(block_hash = %block.block_hash()))]
     async fn process_bitcoin_block(&mut self, block: bitcoin::Block) -> Result<(), Error> {
         tracing::info!("processing bitcoin block");
-        let info = self.stacks_client.get_tenure_info().await?;
+        let tenure_info = self.stacks_client.get_tenure_info().await?;
 
         tracing::debug!("fetching unknown ancestral blocks from stacks-core");
         let stacks_blocks = crate::stacks::api::fetch_unknown_ancestors(
             &self.stacks_client,
             &self.context.get_storage(),
-            info.tip_block_id,
+            tenure_info.tip_block_id,
         )
         .await?;
 
         self.write_stacks_blocks(&stacks_blocks).await?;
         self.write_bitcoin_block(&block).await?;
 
+        // NOTE: unless we add a delay, `block` may still be unknown to the stack node,
+        // so we set the block parent consensus hash.
+        // This also applies to `get_tenure_info`: adding a ~3 sec delay in devenv
+        // returns a highest tenure.
+        self.write_bitcoin_consensus_hash(&block.header.prev_blockhash.into())
+            .await?;
```

Comment:
Do we want to add a delay (in the main loop) to give the stacks node enough time to process the bitcoin block?
Otherwise we will "lag" one bitcoin block behind, and we will probably need to change what stacks tip we fetch (as if we pass the current bitcoin tip, it will have no CH and thus we will get no stacks tip).